paymentsheet/src/test/java/com/stripe/android/paymentsheet/VerticalModePaymentSheetActivityTest.kt

```diff
@@ -143,6 +146,81 @@ internal class VerticalModePaymentSheetActivityTest {
         verticalModePage.assertHasSelectedSavedPaymentMethod("pm_67890")
     }
 
+    @Test
+    fun `Removing card selects next available card`() = runTest(
+        customer = PaymentSheet.CustomerConfiguration(id = "cus_1", ephemeralKeySecret = "ek_test"),
+        networkSetup = {
+            setupElementsSessionsResponse()
+            setupV1PaymentMethodsResponse(count = 2)
+            setupPaymentMethodDetachResponse("pm_12345")
+        },
+    ) {
+        verticalModePage.assertHasSavedPaymentMethods()
+        verticalModePage.assertHasSelectedSavedPaymentMethod("pm_12345")
+        verticalModePage.assertPrimaryButton(isEnabled())
+
+        verticalModePage.clickViewMore()
+        managePage.waitUntilVisible()
+        verticalModePage.assertHasSelectedSavedPaymentMethod("pm_12345")
+        managePage.clickEdit()
+        managePage.clickRemove("pm_12345")
+        managePage.clickDone()
+        Espresso.pressBack()
+
+        verticalModePage.waitUntilVisible()
+        verticalModePage.clickSavedPaymentMethod("pm_67890")
+        verticalModePage.assertPrimaryButton(isEnabled())
+    }
+
+    @Test
+    fun `Removing last card navigates back`() = runTest(
```

Comment:
We already have test coverage of navigating back after removing cards in DefaultManageScreenInteractorTest. I'm not sure we need these tests at all. If we do, what do you think of just having one? In general, I've seen these activity tests be more flaky than the interactor tests which is why I think it's better to just have the test coverage in the interactor tests